lib/IRGen/GenCall.cpp

```diff
@@ -219,6 +219,9 @@ llvm::CallInst *IRGenFunction::emitSuspendAsyncCall(
   auto *id = Builder.CreateIntrinsicCall(llvm::Intrinsic::coro_suspend_async,
                                          {resultTy}, args);
   if (restoreCurrentContext) {
+    // This is setup code after the split point. Don't associate any line
+    // numbers to it.
+    Builder.SetCurrentDebugLocation(llvm::DebugLoc());
```

Comment:
Modifying the state of the builder directly is dangerous because it messes with the expectations of the upper layers. Could you use the ArtificialLocation RAII object in IRGenDebugInfo instead?

Reply:
If it matters that you generate prologue locations instead of line 0, there is also a PrologueLocation RAII object!

Reply:
I wasn't aware we had RAII objects _just_ for debug info! I had, in the past, used the RAII object for storing the insertion point (which also saves the debug loc), but those are not applicable here.

I've updated to use `PrologueLocation`!